### PR TITLE
Add READMEs for Microsoft.Spatial support packages

### PR DESCRIPTION
**Pull request by heaths**

### CODE REVIEW

File: sdk/core/Microsoft.Azure.Core.Spatial.NewtonsoftJson/tests/Samples/Readme.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                Serializer = new NewtonsoftJsonObjectSerializer(serializerSettings)
+            };
+
+            SearchClient client = new SearchClient(endpoint, "mountains", credential, clientOptions);
+            Response<SearchResults<Mountain>> results = client.Search<Mountain>("Rainier");
+
+            foreach (SearchResult<Mountain> result in results.Value.GetResults())
+            {
+                Mountain mountain = result.Document;
+                /*@@*/ _mountain = mountain;
```
Comment (pakrym) [inline sdk/core/Microsoft.Azure.Core.Spatial.NewtonsoftJson/tests/Samples/Readme.cs:74]: I think you can just re-enumerate the results.

**Review by pakrym — APPROVED** [review]
Sweeet! ⛰️🗻⛰️

File: sdk/core/Microsoft.Azure.Core.Spatial.NewtonsoftJson/tests/Samples/Readme.cs (hunk 1 above)
Comment (heaths) [reply sdk/core/Microsoft.Azure.Core.Spatial.NewtonsoftJson/tests/Samples/Readme.cs:74]: I thought about it, but this is cheaper. :) I can change it if you really want me to.

**Review by tg-msft — APPROVED** [review]
Nice

File: sdk/core/Microsoft.Azure.Core.Spatial.NewtonsoftJson/tests/Samples/Readme.cs (hunk 1 above)
Comment (pakrym) [reply sdk/core/Microsoft.Azure.Core.Spatial.NewtonsoftJson/tests/Samples/Readme.cs:74]: I don't like the snippet processor directives, that's all :)

**Pull request merged**